src/Engines/nativeEngine.ts

```diff
@@ -489,6 +494,7 @@ export class NativeEngine extends Engine {
     public setViewport(viewport: Viewport, requiredWidth?: number, requiredHeight?: number): void {
         // TODO: Implement.
         this._cachedViewport = viewport;

```

Comment:
Should we check if the viewport changed? What's the cached value for?

Reply:
Looks like the this._cachedViewPoint is used to get the viewport. So it's fine.

https://github.com/BabylonJS/Babylon.js/blob/master/src/Engines/thinEngine.ts#L394